test/functions/default-arguments/default-argument-class-override3a.future

```diff
@@ -0,0 +1,11 @@
+bug: dynamically dispatched methods with default param arguments wrong
+
+#15164
```

Comment:
In the spirit of closing old issues and having more focused issues, I think we should close #15164 and create a new issue focused specifically on this case. You are welcome to copy my comment https://github.com/chapel-lang/chapel/issues/15164#issuecomment-834260625 into it.

Additionally I'd view adjusting the compiler to emit an error for this case (saying something like "not yet supported") as the immediate next step. Could you note that is the next step in the PR message?

Reply:
Yeah, I had the same plan to close the bigger issue and making a more specific one.

I thought about adding the error message in this PR but (a) had a hard enough time landing this PR's fix without wanting to wrestle with it more and (b) am not convinced that we can actually generate such an error message without either breaking reasonable code or expending a fair amount of effort.  E.g., I think it should be reasonable to continue to use methods on classes that take `param` arguments that take `param` defaults (and believe we rely on such cases heavily), so we'd need to be sure to issue the error only in the case that there was dynamic dispatch involved.  Maybe that's not all that hard, but it seemed like more than a minor tweak to this wrapper code (e.g., I couldn't just check whether the formal we're creating the default for was `param` and issue the error if it is).

Reply:
Yes, it's totally fine to do in a different PR. I suspect the override checking would be the place for it.

Reply:
I spawned off #17683 to capture the remaining part of this issue.  Just to make sure there are no misassumptions, I don't currently have any plans to work on it.